### PR TITLE
user prefs routes: create new, get one, update one

### PR DESCRIPTION
**Pull request by dyoungsmith**

### CODE REVIEW

File: server/db/index.js [hunk 1]
```diff
@@ -10,6 +10,6 @@ const Ingredient = require('./models/ingredient-model.js');
 Ingredient.belongsToMany(Recipe, { through: 'recipes_ingredients', as: 'recipes' });
 Recipe.belongsToMany(Ingredient, { through: 'recipes_ingredients', as: 'ingredients' });
 
-UserPref.hasOne(User);
+UserPref.belongsTo(User);	// UserPref has userId column
```
Comment (johannaperez) [inline server/db/index.js:13]: nice comment so we know how the association works  ;)

File: server/app/routes/user/index.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+});
+
+// Get a particular user loaded with their preferences
+router.get('/:userId/preferences', (req, res, next) => {
+	UserPref.findOne({
+		where: {
+			userId: req.params.userId
+		}
+	})
+	.then(foundPref => {
```
Comment (glebec) [inline server/app/routes/user/index.js:36]: maybe have an `if (!foundPref)` case here? @johanna08 @dyoungsmith

File: server/app/routes/user/index.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+});
+
+// Edit user preferences
+router.put('/:userId/preferences', (req, res, next) => {
+	UserPref.findOne({
+		where: {
+			userId: req.params.userId
+		}
+	})
+	.then(foundPref => {
```
Comment (glebec) [inline server/app/routes/user/index.js:49]: ditto

File: server/db/index.js (hunk 1 above)
Comment (glebec) [reply server/db/index.js:13]: agreed

**Comment by glebec** [comment]
I'm liking how many tests I'm seeing 👍

**Pull request merged**